libvast/src/type.cpp

```diff
@@ -704,6 +704,29 @@ bool compatible(const data& lhs, relational_operator op, const type& rhs) {
   return compatible(rhs, flip(op), lhs);
 }
 
+bool is_superset(const type& x, const type& y) {
+  auto sub = caf::get_if<record_type>(&x);
+  auto super = caf::get_if<record_type>(&y);
+  // If either of the types is not a record type, check if they are
+  // congruent instead.
+  if (!sub || !super)
+    return congruent(x, y);
+  // Check whether all fields of the subset exist in the superset, i.e., no new
+  // fields were added.
```

Comment:
The "no new fields were added" confuses me. It's about checking difference between two given types, no?

Reply:
Ah yes, this comment moved over from when this was inline in the type registry, where we had a new and multiple old types.

Reply:
```suggestion
  // Check whether all fields of the subset exist in the superset.
```